src/floyd-warshall.jl

```diff
@@ -36,17 +36,17 @@ end
 #     Note that it is possible to consume large amounts of memory as the
 #     space required for the FloydWarshallState is O(n^2).
 #     """ ->
-function floyd_warshall_shortest_paths(
+function floyd_warshall_shortest_paths{T<:Real}(
     g::AbstractGraph;
-    edge_dists::AbstractArray{Float64, 2} = Array(Float64,(0,0))
+    edge_dists::AbstractArray{T,2} = zeros(0,0)
 )
 
     use_dists = issparse(edge_dists)? nnz(edge_dists > 0) : !isempty(edge_dists)
     n_v = nv(g)
-    dists = fill(convert(Float64,Inf), (n_v,n_v))
-    parents = zeros(Int, (n_v,n_v))
+    S = typeof(one(T) + one(T))
+    dists = fill(typemax(S), n_v, n_v)
```

Comment:
In 0.4:

```
julia> uint8(1)+uint8(1)
WARNING: uint8(x::Integer) is deprecated, use x % UInt8 instead.
 in depwarn at /usr/local/julia-latest/lib/julia/sys.dylib
 in uint8 at deprecated.jl:29
0x02

julia> uint8(1)+uint8(1) |> typeof
UInt8
```

But I'm confused: is there a reason we want the typemax to be a different type than the number in question?